### PR TITLE
Fix user update dto

### PR DESCRIPTION
**Pull request by matboivin**
Closes: #167

- Remove fields that are not necessary in `src/nestjs-back/src/users/dto/update-user.dto.ts`
- Update users wins, losses, draws and ratio at the end of a game
- Added missing leaderboard in sidebar:
Screenshot at 2022-05-15 12-11-51

### CODE REVIEW

**Comment by Nnevalti** [comment]
Looks good to me !

File: src/nestjs-back/src/users/users.controller.ts [hunk 1]
```diff
@@ -30,19 +30,18 @@ import { editFileName, imageFileFilter } from 'src/utils/upload';
...
 import { IsLoggedInUserGuard } from "./guard/is-logged-in-user.guard";
 import { JwtAuthGuard } from "../auth/guard/jwt-auth.guard";
-import { QueryFailedError } from 'typeorm';
 
 @Controller('users')
 export class UsersController {
   constructor(private readonly usersService: UsersService) {}
 
-  @UseGuards(JwtAuthGuard)
+  // @UseGuards(JwtAuthGuard)
```
Comment (aurelien-brabant) [inline src/nestjs-back/src/users/users.controller.ts]: Can you explain why this shouldn't require authentication?

File: src/nestjs-back/src/users/users.service.ts [hunk 2]
```diff
@@ -186,6 +193,11 @@ export class UsersService {
     let user: User | null = null;
     let tmpDto = {};
 
+    /* Informations */
+    if (updateUserDto.username) {
+      await this.checkUsernameIsAvailable(updateUserDto.username);
```
Comment (aurelien-brabant) [inline src/nestjs-back/src/users/users.service.ts:198]: wording: `isUsernameAvailable` is self descriptive

File: src/nextjs-front/pages/users/[id].tsx [hunk 3]
```diff
@@ -198,14 +198,18 @@ const UserProfilePage: NextPageWithLayout = ({}) => {
...
+      }),
     });
     const reqReceived = await fetch(`/api/users/${id}`, {
       method: "PATCH",
       headers: {
         "Content-Type": "application/json",
       },
-      body: JSON.stringify({ pendingFriendsReceived: [{ id: user.id }] }),
+      body: JSON.stringify({
+        pendingFriendsReceived: [ { "id": user.id } ]
```
Comment (aurelien-brabant) [inline src/nextjs-front/pages/users/[id].tsx:211]: why did you add quotes?

**Review by aurelien-brabant — APPROVED** [review]
LGTM, great job :)

File: src/nestjs-back/src/users/users.controller.ts (hunk 1 above)
Comment (matboivin) [reply src/nestjs-back/src/users/users.controller.ts]: Just saw this and fixed! This was debug

File: src/nestjs-back/src/users/users.service.ts (hunk 2 above)
Comment (matboivin) [reply src/nestjs-back/src/users/users.service.ts]: Renamed!

File: src/nextjs-front/pages/users/[id].tsx (hunk 3 above)
Comment (matboivin) [reply src/nextjs-front/pages/users/[id].tsx]: Removed quotes!

**Pull request merged**